app/src/main/java/app/revanced/integrations/boostforreddit/FixSLinksPatch.java

```diff
@@ -0,0 +1,93 @@
+        File dbFile = context.getDatabasePath("reddit.db");
+        if (!dbFile.exists()) {
+            Logger.printInfo(() -> "Reddit database is not present at " + dbFile.getPath());
+            return null;
+        }
+        SQLiteDatabase redditDb = SQLiteDatabase.openDatabase(
+                dbFile.getPath(),
+                null, SQLiteDatabase.OPEN_READONLY);
+        Cursor tokenCursor = redditDb.rawQuery("SELECT AccessToken FROM User", null);
+        if (tokenCursor.getCount() == 0) {
+            Logger.printInfo(() -> "No authorized users found");
+            return null;
+        }
+        tokenCursor.moveToFirst();
+        String token = tokenCursor.getString(0);
+        tokenCursor.close();
+        redditDb.close();
+        return token;
+    }
+
+    public static String resolveSLink(Context context, String link) {
+        if (link.matches(".*reddit\\.com/r/[^/]+/s/[^/]+")) {
+            Logger.printInfo(() -> "Resolving " + link);
+            try {
+                HttpURLConnection connection = getHttpURLConnection(context, link);
+
+                // Disable strict mode in order to allow network access on the main thread.
+                // This is not ideal, but it's the easiest solution for now.
+                final var currentPolicy = StrictMode.getThreadPolicy();
+                StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+                StrictMode.setThreadPolicy(policy);
+
+                connection.connect();
+                String location = connection.getHeaderField("location");
+                connection.disconnect();
+
+                // Restore the original strict mode policy.
+                StrictMode.setThreadPolicy(currentPolicy);
+
+                Logger.printInfo(() -> "Resolved " + link + " -> " + location);
+                // For some reason using requireNotNull or similar ends up in java.lang.ExceptionInInitializerError,
+                // despite exception being caught down below?
+                if (location == null) {
+                    Logger.printInfo(() -> "Location is null - returning link.");
+                    return link;
+                }
+                return location;
+            } catch (SocketTimeoutException e) {
+                Logger.printInfo(() -> "Reddit request timeout. Censored network?");
+            } catch (Exception e) {
+                Logger.printException(() -> "Failed to resolve " + link, e);
+            }
+        }
+
+        return link;
+    }
+
+    @NonNull
+    private static HttpURLConnection getHttpURLConnection(Context context, String link) throws IOException {
+        URL url = new URL(link);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestProperty("User-Agent", "android:app.revanced.reddit:v1.0 (by /u/spez)");
+        // Auth token is needed if users IP is blacklisted by reddit, e.g. VPN
```

Comment:
Any reason why this isn't necessary in the Sync integrations?

Reply:
This is necessary in Sync as well if you have IP address that is banned by reddit, as far as I understand it - gonna check later. Original PR probably didnt count for that because it was introduced before reddit started banning VPN IPs left and right.

Reply:
Yep - just checked, Sync gets null instead of proper URL on VPN that was blacklisted by reddit:
```05-11 21:46:21.754  2452  2452 I revanced: FixSLinksPatch: Resolving https://www.reddit.com/r/PSP/s/JoTORdHSwS
05-11 21:46:22.556  2452  2452 I revanced: FixSLinksPatch: Resolved https://www.reddit.com/r/PSP/s/JoTORdHSwS -> null```